### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Some phpdoc fixes I've found raising phpstan and psalm levels locally.

I am targeting this branch, because these changes are BC.

## Changelog

```markdown
### Fixed
- Fixed API form handling
```

### CODE REVIEW

File: src/Controller/Api/Legacy/MessageController.php [hunk 1]
```diff
@@ -102,19 +103,19 @@ public function getMessagesAction(ParamFetcherInterface $paramFetcher)
...
 
-        $form = $this->formFactory->createNamed(null, 'sonata_notification_api_form_message', $message, [
+        $form = $this->formFactory->createNamed('', MessageSerializationType::class, $message, [
             'csrf_protection' => false,
         ]);
 
         $form->handleRequest($request);
 
-        if ($form->isValid()) {
+        if ($form->isSubmitted() && $form->isValid()) {
```
Comment (franmomu) [inline src/Controller/Api/Legacy/MessageController.php:118]: I think this wasn't working, passing a form alias to `createNamed` was deprecated some versions ago and I think it was also mandatory in Symfony `4.0` to call `isSubmitted` before `isValid`.

File: src/Backend/QueueDispatcherInterface.php [hunk 2]
```diff
@@ -26,7 +26,7 @@ interface QueueDispatcherInterface
     /**
      * Get a backend by message type.
      *
-     * @param string $type
+     * @param string|null $type
```
Comment (VincentLanglet) [inline src/Backend/QueueDispatcherInterface.php:29]: This method is returning a backend for the corresponding type, so it seems weird to pass `null`.

Why this change ? Maybe the check should be before calling this method instead ?

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/Backend/QueueDispatcherInterface.php (hunk 2 above)
Comment (franmomu) [reply src/Backend/QueueDispatcherInterface.php:29]: I did these changes based on its current use:

https://github.com/sonata-project/SonataNotificationBundle/blob/989b954e5d7fa9b546d21ce2fe92fefab6927a60/src/Command/CleanupCommand.php#L47

https://github.com/sonata-project/SonataNotificationBundle/blob/989b954e5d7fa9b546d21ce2fe92fefab6927a60/src/Backend/MessageManagerBackendDispatcher.php#L61-L67

**Review by VincentLanglet — DISMISSED** [review]

File: src/Backend/QueueDispatcherInterface.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Backend/QueueDispatcherInterface.php:29]: We could add a comment maybe ?
And it would be nice to change the signature to `public function getBackend($type = null)` but it's BC, so we can only add a NEXT_MAJOR comment.

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks
